File: core/src/client/ImageManager.scala
package client

import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.Gdx;
import javax.imageio.ImageIO
import java.io.File
import network.client.GameConnection
import engine.rts.model.Resource
import java.util._
import java.awt.image.BufferedImage
import scala.collection.JavaConversions._

/**
 * This class is responsible for loading images and storing them.
 */
class ImageManager(val serverConnection:GameConnection) {

    val imageDir="images/"
    val combatImageDir="images/combat/"
    val flagDir="images/flags/"
    val configLoc="mapData/TestMap/"

    val armyImage=new Texture(combatImageDir+"troop.png")
    val siegeImage=new Texture(combatImageDir+"fire.jpg")
    val battleImage=new Texture(combatImageDir+"battle.png")
    val mapBackground=new Texture(imageDir+"MapBackground.jpg")
    val capitalImage=new Texture(imageDir+"MapBackground.jpg")

    var resourceImages=new ArrayList[Texture]
    var improvementImages=new ArrayList[Texture]
    
    readImages()

    def getFlag(pName:String)=flagDir+pName+".jpg"
    def unreadImages:Boolean=resourceImages==null
    def getUpgradeImage(uNum:Int)=improvementImages.get(uNum)
    def getResourceImage(rNum:Int)=resourceImages.get(rNum)
       
    /**
     * This method reads the images that represent the resources and improvements.
     */
    private def readImages() {

        val resourceData:ArrayList[Resource]=serverConnection.getResourceDefs()
        try {
            resourceData.foreach {resource:Resource=>
            	val resourceImage = new Texture("images/resources/" + resource.getResourceFile())
            	resourceImages.add(resourceImage)
            }
            serverConnection.improvementDefs().foreach {improvement=>
                val upgradeImage = new Texture("images/improvements/" + improvement.getImageLocation())
                improvementImages.add(upgradeImage)
            }
        } 
        catch {
            case e: Exception => e.printStackTrace()
        }
    }
}

File: core/src/client/ImageManager.java
package client;

import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.Gdx;

import javax.imageio.ImageIO;

import server.model.UpgradeDefinition;

import java.io.File;
import java.util.ArrayList;

import network.client.GameConnection;
import engine.rts.model.Resource;

import java.awt.image.BufferedImage;

/**
 * This class is responsible for loading images and storing them.
 */
public class ImageManager{

    String imageDir="images/";
    String combatImageDir="images/combat/";
    String flagDir="images/flags/";
    String configLoc="mapData/TestMap/";

    public final Texture armyImage=new Texture(combatImageDir+"troop.png");
    public final Texture siegeImage=new Texture(combatImageDir+"fire.jpg");
    public final Texture battleImage=new Texture(combatImageDir+"battle.png");
    public final Texture mapBackground=new Texture(imageDir+"MapBackground.jpg");
    public final Texture capitalImage=new Texture(imageDir+"MapBackground.jpg");

    ArrayList<Texture> resourceImages=new ArrayList<Texture>();
    ArrayList<Texture> improvementImages=new ArrayList<Texture>();
    
	final GameConnection serverConnection;
	
	public ImageManager(GameConnection sCon){
		serverConnection=sCon;
	    readImages();

	}
	
    public String getFlag(String pName){
    	return flagDir+pName+".jpg";
    }
    
    public boolean unreadImages(){
    	return resourceImages==null;
    }
    
    public Texture getUpgradeImage(int uNum){
    	return improvementImages.get(uNum);
    }
    
    public Texture getResourceImage(int rNum){
    	return resourceImages.get(rNum);
    }
       
    /**
     * This method reads the images that represent the resources and improvements.
     */
    private void readImages() {

        ArrayList<Resource> resourceData=serverConnection.getResourceDefs();
        try {
        	for(Resource resource:resourceData){
            	Texture resourceImage = new Texture("images/resources/" + resource.getResourceFile());
            	resourceImages.add(resourceImage);
            }
            for(UpgradeDefinition improvement:serverConnection.improvementDefs()){
                Texture upgradeImage = new Texture("images/improvements/" + improvement.getImageLocation());
                improvementImages.add(upgradeImage);
            }
        } 
        catch(Exception e) {
            e.printStackTrace();
        }
    }
}